server/settings/main.py

```diff
@@ -1,67 +1,188 @@
-from pydantic import Field
+from ayon_server.entities.core.attrib import attribute_library
+from ayon_server.settings import BaseSettingsModel, SettingsField
+from ayon_server.settings.enum import secrets_enum, anatomy_presets_enum
 
-from ayon_server.settings import BaseSettingsModel
-from ayon_server.settings.enum import secrets_enum
-from ayon_server.graphql.resolvers.users import get_users
+
+def default_shotgrid_entities():
+    """The entity types enabled in ShotGrid.
+
+    Return a list to be consumed by the `enum_resolver` in
+    `ShotgridCompatibilitySettings.shotgrid_enabled_entities`.
+    """
+    return [
+        "Project",
+        "Episode",
+        "Sequence",
+        "Scene",
+        "Shot",
+        "Asset",
+        "Task",
+    ]
+
+
+def get_default_folder_attributes():
+    """Get AYON's Folder attributes
+
+    Get all the `attribs` for Folder entities in a list
+    to be consumed by the `default_factory` in the
+    `ShotgridCompatibilitySettings.custom_attribs_map`
+    settings.
+    """
+    attributes = []
+
+    for attr_dict in attribute_library.data.get("folder", {}):
+        attr_name = attr_dict["name"]
+
+        if attr_name in ["shotgridId", "shotgridType", "tools"]:
+            continue
+
+        attr_map = {
+            "ayon": attr_name,
+            "sg": "",
+            "type": [attr_dict["type"]],
+            "scope": default_shotgrid_entities()
+        }
+
+        if attr_map not in attributes:
+            attributes.append(attr_map)
+
+    return attributes
 
 
 class ShotgridServiceSettings(BaseSettingsModel):
-    """Specific settings for the Shotgrid Services: Processor, Leecher and Transmitter.
+    """Specific settings for services
+
+    ShotGrid Services: Processor, Leecher and Transmitter.
 
-   The different services process events from either Shotgrid or AYON, this field
-   allows to control how long to wait between each event is processed.
+   The different services process events from either ShotGrid or AYON,
+   this field allows to control how long to wait between each event
+   is processed.
     """
-    polling_frequency: int = Field(
+    polling_frequency: int = SettingsField(
         default=10,
-        title="How often (in seconds) to process Shotgrid related events.",
+        title="How often (in seconds) to process ShotGrid related events.",
         validate_default=False,
     )
 
-    ayon_service_user: str = Field(
-        default="service",
-        title="AYON service user",
-        description="The AYON user used in the services (the user corresponding to the `AYON_API_KEY` set in the service)",
+
+class AttributesMappingModel(BaseSettingsModel):
+    _layout = "compact"
+    ayon: str = SettingsField(title="AYON")
+    sg: str = SettingsField(title="SG")
+    # TODO: how do you make this a single selectable entry
+    type: list[str] = SettingsField(
```

Comment:
we can remove the `type` from the exposed settings as suggested by @iLLiCiTiT so you can't choose it and it simply infers the type from the Ayon attribute. That way I wouldn't need to get the answer to my `TODO` of how to make the enum be a single entry selection and not a list

Reply:
Here is an example on how to do it - just for the future reference @fabiaserra :

```
def tile_assembler_enum():
    """Return a list of value/label dicts for the enumerator.

    Returning a list of dicts is used to allow for a custom label to be
    displayed in the UI.
    """
    return [
        {
            "value": "DraftTileAssembler",
            "label": "Draft Tile Assembler"
        },
        {
            "value": "OpenPypeTileAssembler",
            "label": "Open Image IO"
        }
    ]
    
tile_assembler_plugin: str = SettingsField(
        title="Tile Assembler Plugin",
        enum_resolver=tile_assembler_enum,
    )
```

Reply:
addressed in [`3f8b6d3` (#72)](https://github.com/ynput/ayon-shotgrid/pull/72/commits/3f8b6d38ffb0d3dad871b8f720302a490c4df829)

Reply:
also reviewed the code and realized that `type` has to be returned [e547ab7](https://github.com/ynput/ayon-shotgrid/pull/72/commits/e547ab72214c6627cc8b8cc9024a79603a0d074b)

Reply:
Well in order to remove it you'd need to delegate the type inference at the moment that the SG attributes are created or during the creation of the attributes dictionary

Reply:
I had created the field but disable its editablity https://github.com/ynput/ayon-shotgrid/pull/72/commits/ff8e5935eb592c3d1582c430d9b8fe33c8f59746

![image](https://github.com/ynput/ayon-shotgrid/assets/40640033/e3eff8da-5d11-4c45-b04e-4a0eeee1ecc3)

Reply:
sounds good, it's actually good for visibility so it's clear to the user that the created field in AYON fits what you'd expect

Reply:
it's also a good proof that the AYON attribute you type actually exists, if it doesn't exist the field type would error out

Reply:
But this approach is having its downfall - if anybody will need to add new field, he will not be able to edit its type. Which I am assuming will never happen. But probably more safe it would be to resolve attribute types somewhere in  `get_or_create_sg_field`

Reply:
types would be edited from the AYON attributes settings? 
![image](https://github.com/ynput/ayon-shotgrid/assets/4348536/852509d8-4da0-4453-8b6b-9a0c4150116d)

Or you mean once that is set and the SG field gets created, editing it later wouldn't propagate to SG? that would have been the case prior to this as well so the solution is that the Admin would need to delete the SG fields prior to the edit and the re-sync